src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -432,6 +433,26 @@ CHIP_ERROR DiscoveryImplPlatform::ResolveNodeId(const PeerId & peerId, Inet::IPA
 {
     ReturnErrorOnFailure(Init());
 
+    FabricId fabricId;
+    Inet::IPAddress addr;
+    uint16_t port;
+    Inet::InterfaceId iface;
+
+    if (sIPCache.Lookup(peerId.GetNodeId(), fabricId, addr, port, iface) == CHIP_NO_ERROR)
+    {
+        ResolvedNodeData nodeData;
+
+        nodeData.mInterfaceId = iface;
+        nodeData.mPort        = port;
+        nodeData.mAddress     = addr;
+        nodeData.mPeerId.SetNodeId(peerId.GetNodeId());
+        nodeData.mPeerId.SetFabricId(fabricId);
```

Comment:
Fabric ID is always insufficient.

The full operational identity of node is FabricRootPublicKey + Fabric Id + NodeId.

In operational discovery, the "compressed fabric reference" is used in service records, to go from 73 bytes to 8 bytes.

Since on a local device, the adminId can map directly to FabricRootPublicKey + Fabric Id (the 73 bytes value) and/or compressed fabric reference, I suggest the mapping use the adminId.

Here, for the next call, it looks like you want the to go from "compressed fabric reference" of a DNS-SD operational discovery record to an underlying adminId. 

I feel like it would be a good time to somewhat refactor some of this to renaming things clearly, since there isn't actually a fabric ID in Operational Discovery (i.e. `_matter._tcp`) service record.

Reply:
@tcarmelveilleux you are correct in terms of unique identity. Unfortunately our uniqueness is currently not spec-compliant in the SDK. We have nodeid in several places (@pan-apple - I think most of our connection tracking is using nodeid and needs to be moved to peerid) and peerid is obviously incomplete.

I would suggest to keep this PR as is and not hinge it on root key support. The full spec for multi-admin is supposed to be built separately (@shana-apple , @woody-apple ?) and I expect that to include the full peer id update to be actually unique.

As long as this PR uses peerid, once we make peerid to be spec it should flag all areas that need fixing.

Reply:
> @bitkis I have two general questions before digging deeper into the change:
> 
> 
>     2. Would it make sense to make your cache optional (controlled with a compile-time flag)? I wonder if platforms that already provide their own cache (like Avahi?) need another layer of caching or maybe it would be safer to make it optional?

I was thinking the same thing.   I wonder how this plays with avahi?  I wonder if should be "internal cache"/"external cache (avahi)" or "no cache".  The code is small, should it be run-time selection or compile time selection?